### PR TITLE
sentieon-genomics: add version 202112.02, remove manual download

### PR DESCRIPTION
**Pull request by snehring**
They document the path in their dockerfile recipe, I think we can include it here.

### CODE REVIEW

File: var/spack/repos/builtin/packages/sentieon-genomics/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/sentieon-genomics/package.py:26]: Can you point me where you got this download link? I checked sentieon home page and couldn't find it.

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/sentieon-genomics/package.py
Comment (snehring) [reply var/spack/repos/builtin/packages/sentieon-genomics/package.py:26]: https://github.com/Sentieon/sentieon-docker/blob/master/Dockerfile

Comment (alalazo) [reply var/spack/repos/builtin/packages/sentieon-genomics/package.py:26]: @snehring Thanks, can you add a comment on top of the `url` saying we are using the permalink published in their Dockerfile and adding the link above?

@donFreed I assume the answer is yes, but just to double check, would it be fine to redistribute the binaries from:
```console
https://s3.amazonaws.com/sentieon-release/software/sentieon-genomics-201808.01.tar.gz
```
which was extracted from your public Dockerfile?

Comment (DonFreed) [reply var/spack/repos/builtin/packages/sentieon-genomics/package.py:26]: Yes, redistribution of the precompiled binaries is OK.
Just be aware that we are a commercial software company and the software requires an active software license before it can be used.

Comment (alalazo) [reply var/spack/repos/builtin/packages/sentieon-genomics/package.py:26]: Thanks all for the clarifications.

**Review by alalazo — DISMISSED** [review]

File: var/spack/repos/builtin/packages/sentieon-genomics/package.py
Comment (tgamblin) [reply var/spack/repos/builtin/packages/sentieon-genomics/package.py:26]: Is this in the EULA?  It would be nice to be able to point to something that says we can do this, but this comment may do if there isn't anything.

File: var/spack/repos/builtin/packages/sentieon-genomics/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/sentieon-genomics/package.py:26]:
```suggestion
    # See also: https://github.com/spack/spack/pull/30145/files#r853275635
    url      = "https://s3.amazonaws.com/sentieon-release/software/sentieon-genomics-201808.01.tar.gz"
```

Comment (alalazo) [reply var/spack/repos/builtin/packages/sentieon-genomics/package.py:26]: @snehring Apologies, can you quickly add this? I'll merge asap after that.

File: var/spack/repos/builtin/packages/sentieon-genomics/package.py
Comment (DonFreed) [reply var/spack/repos/builtin/packages/sentieon-genomics/package.py:26]: Here is the text of our EULA, https://www.sentieon.com/EULA/eula-aws.html. It would be great if this could somehow be pointed to in spack.

The EULA expressly forbids copying without prior approval. Hopefully the above comment is sufficient?

Comment (snehring) [reply var/spack/repos/builtin/packages/sentieon-genomics/package.py:26]: @tgamblin would you like me to include a link to the EULA or revert to manual download?

Comment (snehring) [reply var/spack/repos/builtin/packages/sentieon-genomics/package.py:26]: I've pushed a commit that adds a link to the eula. @tgamblin let me know if this is satisfactory.

File: var/spack/repos/builtin/packages/sentieon-genomics/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/sentieon-genomics/package.py:15]: :heavy_check_mark:

File: var/spack/repos/builtin/packages/sentieon-genomics/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/sentieon-genomics/package.py:25]: :heavy_check_mark:

**Review by alalazo — APPROVED** [review]
We have both a link to the EULA and the comment from @DonFreed so I'll be merging this

**Pull request merged**
